### PR TITLE
Add typed UI geometry context

### LINKED ISSUE #820: [Feature]: Add typed UI geometry context and correct JSON documentation
### Is your feature request related to a problem? Please describe.

Parent: #798

UIA BoundingRectangle values are physical screen pixels, while WinUI layout values are device-independent pixels (DIPs). Callers currently lack the target window's DPI context for conversion.

`inspect` and `search` already expose typed numeric element geometry, but `get-property --json` converts values such as BoundingRectangle and IsOffscreen to strings. Some shipped JSON documentation also says `controlType` even though the published element model emits `type`.

### Describe the solution you'd like

Add explicit, additive geometry context without removing published fields:

- On each `inspect --json` `windows[]` entry, and on the target reported by `status --json`, add:
  - `windowDpi`: the value returned by `GetDpiForWindow(hwnd)`
  - `scale`: `windowDpi / 96`
  - `dpiAwareness`: unaware, system-aware, or per-monitor-aware
  - `coordinateSpace`: `physical-screen-pixels`
- Describe these as the window's DPI context, not unconditional monitor DPI. `GetDpiForWindow` returns 96 for unaware windows, system DPI for system-aware windows, and monitor DPI for per-monitor-aware windows.
- Add a scrubbed, typed `element` object to `get-property --json`, while preserving its existing `elementId` and string-valued `properties` map.
- Preserve existing numeric `x`, `y`, `width`, and `height`.
- Document that `0,0,0,0` is UIA's empty/no-displayed-UI rectangle in this COM projection. Add no redundant bounds marker.
- Keep `isOffscreen` independent because an offscreen element may still have nonzero bounds.
- Keep the published `type` field canonical and correct documentation/examples that currently say `controlType`.
- Document the actual `inspect`, `search`, `wait-for`, and `get-property` envelopes.

### Additional context

Validate the contract on mixed-DPI monitors with a per-monitor-aware WinUI window and popup HWND on different monitors. Invalid HWND/DPI reads must be surfaced explicitly rather than silently defaulting to 96.

### PR DESCRIPTION
**Pull request by nmetulev**
## Summary
- add target-window DPI context to `ui inspect --json` windows and `ui status --json`
- add a scrubbed typed `element` to `ui get-property --json` while preserving `elementId` and string-valued `properties`
- document the canonical JSON envelopes, physical-pixel geometry semantics, and `type` field

## Validation
- targeted UI/DPI tests: 605 passed, 1 opt-in interactive test skipped
- live `ui status --json` against a per-monitor-aware window: 120 DPI / 1.25 scale
- live compatibility check against a process with no top-level window: succeeds with `hwnd: 0` and omitted DPI fields
- `scripts/build-cli.ps1 -SkipTests`: passed, including NativeAOT publish, npm package, NuGet package, and MSIX packaging
- `scripts/validate-plugin-package.ps1`: passed
- full `scripts/build-cli.ps1`: compilation/npm/analyzer phases passed; test phase had three unrelated environment-sensitive failures in existing crash-dump/recording tests

Mixed-monitor placement could not be exercised on the available desktop; distinct per-window DPI/awareness values are covered by command tests.

Fixes #820

### CODE REVIEW

**Comment by zateutsch** [comment]
## PR Review — typed UI geometry context

**Decision: changes required.** The feature is well-scoped, on-mission, documented, and green (builds clean with 0 warnings, 68 targeted tests pass). One real regression blocks it: `ui inspect --json` now hard-fails for a window whose HWND is `0` — a state the rest of the code (and this same PR's `ui status` path) treats as normal.

### 🔴 Must fix — `ui inspect --json` crashes on a window with no HWND

- **What is wrong:** The new DPI loop calls `windowDpiContextProvider.GetForWindow(windowInfo.Hwnd)` for **every** window unconditionally. `BuildWindows` can emit a window with `Hwnd == 0` (it uses `el.WindowHandle ?? uiTarget.WindowHandle` / `?? 0`, and `Flush()` deliberately keeps a zero-HWND window when it has element roots). `WindowDpiContextProvider.GetForWindow(0)` throws `InvalidOperationException("...HWND is zero")`, so the whole inspect fails with exit 1. `ui status` avoids this with a `uiTarget.WindowHandle != 0` guard — inspect has no such guard.
- **Show me:** `winapp ui inspect -a <process resolved before it has a top-level window, or a windowless UIA root> --json` → exits 1 with `{"error":{...,"message":"Cannot read the target window DPI context because its HWND is zero."}}`; expected: the element tree still serializes, with DPI fields omitted for that window (exactly what `ui status --json` does for the same state).
- **Why it matters:** Optional DPI metadata turns the command's primary result (the UI tree) into a hard failure. Existing automation on the PID/windowless path loses `inspect --json` entirely. The added `Status_Json_ProcessWithoutWindow` test proves the team already considers HWND `0` a supported, non-error state.
- **Smallest fix:** Mirror `ui status`: skip DPI enrichment when `windowInfo.Hwnd == 0` and make the inspect window's DPI fields nullable/omitted for that case. Add an inspect test with a zero-HWND window — the current `FakeWindowDpiContextProvider` returns a value for HWND `0` instead of throwing like the real provider, so it masks this path.
- **Location:** `src/winapp-CLI/WinApp.Cli/Commands/UiInspectCommand.cs` (the DPI enrichment loop, ~L147-153)

### 🟡 Non-blocking — `get-property` typed `element` includes `value` regardless of `--property`

- **What is wrong:** The new typed `element` serializes the full `UiElement`. `UiElementScrubber.Strip` clears only `Id`/`ParentSelector`/`WindowHandle`, never `Value` (editable-field text from UIA `ValuePattern`). So `get-property <field> --property IsEnabled --json` scopes the string `properties` map to `IsEnabled` but still emits the field's text in `element.value`.
- **Show me:** `winapp ui get-property <edit-selector> -a <app> --property IsEnabled --json` → output includes `element.value: "<field text>"`; a caller scoping to one benign property wouldn't expect the field content.
- **Why it matters:** Places arbitrary app field text into JSON a script may log. Bounded, and consistent with shipped `ui search`/`ui wait-for` (which already emit `value`) — so this is a conscious-decision call, not a clear defect.
- **Smallest fix:** In `get-property`, clear `element.Value` unless `--property` is `Value` or omitted; or accept it as intended and note it in the reference doc.
- **Location:** `src/winapp-CLI/WinApp.Cli/Commands/UiGetPropertyCommand.cs` (~L97-106), `src/winapp-CLI/WinApp.Cli/Helpers/UiElementScrubber.cs` (`Strip`, ~L44-48)

### What was exercised

- `dotnet build WinApp.Cli.csproj -c Debug` (branch worktree, cold restore) — **succeeded, 0 warnings / 0 errors**.
- `dotnet test` filtered to DPI/status/get-property — **68 passed, 0 failed** (offline; no network needed).
- Static trace of `BuildWindows` → inspect DPI loop → `WindowDpiContextProvider.GetForWindow` — confirmed the zero-HWND crash chain; the real provider's throw-on-`0` is proven by `GetForWindow_ZeroHwnd_ThrowsExplicitError`.
- **Not exercised:** a live windowless/zero-HWND inspect target — hard to construct deterministically. The finding rests on the code asymmetry with `ui status` plus the real provider's unit-proven throw, not a live repro, so the *frequency* in the wild is unquantified while the failure path itself is certain.

Necessity and ship-surfaces reviews returned clean (docs are consistent and complete). A separate different-model-family cross-check confirmed all findings and added none. A low "centralize the 4-field DPI copy into one helper" suggestion was dropped as no user impact.

<sub>🤖 Generated with multi-dimensional PR review</sub>

**Review by zateutsch — COMMENTED** [review]
One must fix surfaced above, it seems like the trigger circumstances would be pretty rare (an inspect before a window loads) but since it would cause a crash it might be worth addressing.

Up to your judgement.

File: src/winapp-CLI/WinApp.UIAutomation/Services/UiAutomationService.cs [hunk 1]
```diff
@@ -204,66 +210,110 @@ public Task<UiElement[]> InspectAsync(UiTarget uiTarget, string? elementId, int
...
-                {
-                    Id = $"--- HWND {mainHwnd}",
-                    Type = "---",
-                    Name = $"HWND {mainHwnd}: \"{mainTitle}\" ({mainInfo.Label}, {mainInfo.ClassName})",
-                    Depth = 0,
-                    WindowHandle = mainHwnd
-                });
+                promotableWindowHandles.Add(window.Hwnd);
             }
+        }
```
[the remark this answers is not in the document]
Comment (nmetulev) [reply src/winapp-CLI/WinApp.UIAutomation/Services/UiAutomationService.cs]: Updated the loop to filter `independentWindows` with `Where(window => window.IsInSelectedTree)` before adding promotable HWNDs. The focused inspect suite passes after the change.

File: src/winapp-CLI/WinApp.UIAutomation/Services/UiAutomationService.cs [hunk 2]
```diff
@@ -204,66 +210,107 @@ public Task<UiElement[]> InspectAsync(UiTarget uiTarget, string? elementId, int
...
-
-        // Set WindowHandle on all elements from main window
-        foreach (var el in elements)
-        {
-            el.WindowHandle = uiTarget.WindowHandle;
-        }
-
-        // Also walk popup/owned windows (when inspecting full tree, not scoped to element,
-        // and the user did not explicitly target a single HWND — see issue #472).
+        // A process-target inspect must keep each top-level HWND in its own output group. UIA can
```
[the remark this answers is not in the document]
Comment (nmetulev) [reply src/winapp-CLI/WinApp.UIAutomation/Services/UiAutomationService.cs:221]: Fixed scoped inspect to replace the session HWND with the resolved element's top-level HWND before walking and stamping the subtree. Added a real owned-window regression verifying `InspectAsync(..., "btnOwned", ...)` carries the owned HWND; the focused service and command tests pass.

File: src/winapp-CLI/WinApp.UIAutomation/Services/UiAutomationService.cs [hunk 3]
```diff
@@ -2034,6 +2123,24 @@ private static UiElement ToUiElement(IUIAutomationElement element, string path,
...
+            var root = global::Windows.Win32.PInvoke.GetAncestor(
+                native,
+                global::Windows.Win32.UI.WindowsAndMessaging.GET_ANCESTOR_FLAGS.GA_ROOT);
+            return root.IsNull ? (nint)native : (nint)root;
+        }
+        catch (COMException)
+        {
+            return 0;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (nmetulev) [reply src/winapp-CLI/WinApp.UIAutomation/Services/UiAutomationService.cs:2169]: Fixed windowless scoped elements by walking control-view ancestors until a native handle is found, then normalizing it with `GetAncestor(GA_ROOT)`. The same helper now backs scoped inspect, ancestor inspect, and capture targeting. Added a real owned `ToolStripMenuItem` regression (NativeWindowHandle 0) selected by Name; it fails before the product fix and passes after it.

**Review by zateutsch — APPROVED** [review]

**Review by zateutsch — APPROVED** [review]
